### PR TITLE
Eliminate connection state listener / stateful connector (first pass)

### PR DESCRIPTION
**Pull request by louis89**
This is only the first pass to completely eliminating the connection state listener this removes it completely from the client connector and prevents the user of this code from needing to deal with it on the server. Also removes the StatefulConnector interface. Obviously the examples still need some work though this proves the concept.

### CODE REVIEW

File: cf-tcp-helloworld-client/src/main/java/org/eclipse/californium/examples/GETClient.java [hunk 1]
```diff
@@ -50,39 +52,38 @@ public static void main(String args[]) throws NoSuchAlgorithmException, Executio
...
+		}
+
 		final GETClient client = new GETClient(args[0]);
 
 	}
-		
-	private TCPEndpoint tcpClientEndpoint;
-	
+
+	private TCPServerEndpoint tcpClientEndpoint;
```
Comment (asummers) [inline cf-tcp-helloworld-client/src/main/java/org/eclipse/californium/examples/GETClient.java]: Server + client

File: cf-tcp-helloworld-server/src/main/java/org/eclipse/californium/examples/HelloWorldServer.java [hunk 2]
```diff
@@ -28,91 +28,93 @@
...
+
+	/*
+	 * Application entry point.
+	 */
+	public static void main(String[] args) throws KeyManagementException, SSLException, NoSuchAlgorithmException {
+
+		try {
+
+			if(args == null || args.length == 0) {
+				System.out.println("No resource to expose was specifed, exposing default value \"TEMP\"");
```
Comment (asummers) [inline cf-tcp-helloworld-server/src/main/java/org/eclipse/californium/examples/HelloWorldServer.java]: Specified

File: cf-tcp-helloworld-server/src/main/java/org/eclipse/californium/examples/HelloWorldServer.java [hunk 3]
```diff
@@ -28,91 +28,93 @@
...
+			// set resource identifier
+			super(id);
+
+			// set display name
+			getAttributes().setTitle("Hello-World Resource");
+		}
+
+		@Override
+		public void handleGET(final CoapExchange exchange) {
+			System.out.println("GET Revceived: " + exchange.toString());
```
Comment (asummers) [inline cf-tcp-helloworld-server/src/main/java/org/eclipse/californium/examples/HelloWorldServer.java]: Received

File: cf-tcp-helloworld-server/src/main/java/org/eclipse/californium/examples/HelloWorldServer.java [hunk 4]
```diff
@@ -28,91 +28,93 @@
...
+	}
+
+	/*
+	 * Constructor for a new Hello-World server. Here, the resources
+	 * of the server are initialized.
+	 */
+	public HelloWorldServer(final String... name) throws SocketException {
+
+		// provide an instance of a Hello-World resource
+		for(final String s : name) {
```
Comment (asummers) [inline cf-tcp-helloworld-server/src/main/java/org/eclipse/californium/examples/HelloWorldServer.java]: Who is s? s is a resource?

File: californium-core/src/main/java/org/eclipse/californium/core/network/tcp/TCPServerEndpoint.java [hunk 5]
```diff
@@ -10,44 +9,40 @@
...
 
-/**
- * This class is meant for server TCP connection only
- *
- * If you build a TCP server that must serve to multiple TCP Clients containing the coapServer (hence multiple CoAP Client from 1 TCP connection)
- * You must use this registry to keep track of your clients. Once a Connection is established, the call will be made for you to create a CoAP client
- * if you wish to do so.
- */
-public abstract class CoapClientRegistry implements ConnectionStateListener {
+public class TCPServerEndpoint extends CoAPEndpoint implements ConnectionStateListener {
```
Comment (joshlarson) [inline californium-core/src/main/java/org/eclipse/californium/core/network/tcp/TCPServerEndpoint.java]: Minor naming point: Most of the rest of the TCP classes are named "TcpBlahBlah", not "TCPBlahBlah". Does anyone mind if I change that?

**Pull request merged**